pkg/satellite/stream/stream.go

```diff
@@ -108,8 +108,12 @@ func (ss *satelliteStream) recvLoop() {
 			} else {
 				log.Fatalf("Error reading from API stream: %v\n", err)
 			}
-		} else {
-			ss.streamId = res.StreamId
+		}
+
+		ss.streamId = res.StreamId
+
+		// Telemetry data can be nil when a StreamEvent comes.
+		if res.GetReceiveTelemetryResponse() != nil {
```

Comment:
Do the generated stubs have an enum for the oneof? It's better to use a switch on the case for languages that support it.